doc/reference/configuration/configuration_reference.rst

```diff
@@ -3,13 +3,280 @@
+
 This topic describes all :ref:`configuration parameters <configuration>` provided by Tarantool.
 
+
+..  _configuration_reference_config_parameter:
+
+<config_parameter>
+------------------
+
+Any configuration parameter that can be defined in the global :ref:`scope <configuration_scopes>`.
+For example, :ref:`iproto <configuration_reference_iproto>` and :ref:`database <configuration_reference_database>` configuration parameters defined at the global level are applied to all instances in the cluster.
+
+
+..  _configuration_reference_groups:
+
+groups
+------
+
+The ``groups`` section provides the ability to define the :ref:`full topology of a Tarantool cluster <configuration_overview>`.
+
+.. NOTE::
+
+    ``groups`` can be defined in the global :ref:`scope <configuration_scopes>` only.
+
+-   :ref:`groups.\<group_name\> <configuration_reference_groups_name>`
+-   :ref:`groups.\<group_name\>.replicasets <configuration_reference_groups_name_replicasets>`
+-   :ref:`groups.\<group_name\>.\<config_parameter\> <configuration_reference_groups_name_config_parameter>`
+
+.. _configuration_reference_groups_name:
+
+.. confval:: groups.<group_name>
+
+    A group name.
+
+
+.. _configuration_reference_groups_name_replicasets:
+
+.. confval:: groups.<group_name>.replicasets
+
+    Replica sets that belong to this group. See :ref:`replicasets <configuration_reference_replicasets>`.
+
+
+.. _configuration_reference_groups_name_config_parameter:
+
+.. confval:: groups.<group_name>.<config_parameter>
+
+    Any configuration parameter that can be defined in the group :ref:`scope <configuration_scopes>`.
+    For example, :ref:`iproto <configuration_reference_iproto>` and :ref:`database <configuration_reference_database>` configuration parameters defined at the group level are applied to all instances in this group.
+
+
+
+..  _configuration_reference_replicasets:
+
+replicasets
+~~~~~~~~~~~
+
+.. NOTE::
+
+    ``replicasets`` can be defined in the group :ref:`scope <configuration_scopes>` only.
+
+-   :ref:`replicasets.\<replicaset_name\> <configuration_reference_replicasets_name>`
+-   :ref:`replicasets.\<replicaset_name\>.leader <configuration_reference_replicasets_name_leader>`
+-   :ref:`replicasets.\<replicaset_name\>.bootstrap_leader <configuration_reference_replicasets_name_bootstrap_leader>`
+-   :ref:`replicasets.\<replicaset_name\>.instances <configuration_reference_replicasets_name_instances>`
+-   :ref:`replicasets.\<replicaset_name\>.\<config_parameter\> <configuration_reference_replicasets_name_config_parameter>`
+
+.. _configuration_reference_replicasets_name:
+
+.. confval:: replicasets.<replicaset_name>
+
+    A replica set name.
+
+
+.. _configuration_reference_replicasets_name_leader:
+
+.. confval:: replicasets.<replicaset_name>.leader
+
+    A replica set leader.
+    This option is in effect when ``manual`` :ref:`replication.failover <configuration_reference_replication_failover>` is used.
```

Comment:
No `leader` or `leader: null` is an important case. A usual master-replica switch process assumes a short period, when all the instances are in RO.

A correct master switch in the `failover: manual` mode would involve the following steps:

1. Remove the `leader` key or set it to `null`.
2. Wait until all the data goes to the future master (compare vclock values).
3. Set the new leader using the `leader` key.

(BTW, things are pretty same for the `failover: off` mode if a single master at time is assumed.)

Reply:
Thanks for the clarification - very useful info for the upcoming replication tutorials. Will update reference too.